util/ratelimits.go

```diff
@@ -58,25 +58,30 @@ func (c *PostRequestCounter) ApiResponseRatelimitCounter(resp *resty.Response) e
 	if err != nil {
 		return err
 	}
+
+	// TODO: remove when rate-limits are simplified
+	currTime := time.Now().Unix()
+	secondaryLimitRefreshTime := c.RefreshTime - int(reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay.Seconds())
+	if c.ReqRemaining >= reconciler.SecondaryPOSTRequestLimit && currTime <= int64(secondaryLimitRefreshTime) {
+		c.RefreshTime = secondaryLimitRefreshTime
+	}
 	return nil
 }
 
 // IsPOSTLimitReached checks whether POST limits have been reached.
 func (c *PostRequestCounter) IsPOSTLimitReached() bool {
 	// TODO: Once linode API adjusts rate-limits, remove secondary rate limit and simplify accordingly
 	currTime := time.Now().Unix()
+	return currTime <= int64(c.RefreshTime) && (c.ReqRemaining == 0 || c.ReqRemaining == reconciler.SecondaryPOSTRequestLimit)
+}
 
-	if c.ReqRemaining == 0 {
-		if currTime <= int64(c.RefreshTime) {
-			return true
-		}
-	}
-
-	secondaryLimitRefreshTime := c.RefreshTime - int(reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay.Seconds())
-	if c.ReqRemaining <= reconciler.SecondaryPOSTRequestLimit && currTime <= int64(secondaryLimitRefreshTime) {
-		return true
+// RetryAfter returns how long to wait in seconds for rate-limit to reset
+func (c *PostRequestCounter) RetryAfter() time.Duration {
+	currTime := time.Now().Unix()
+	if currTime > int64(c.RefreshTime) {
+		return 0
 	}
-	return false
+	return time.Duration(c.RefreshTime-int(currTime)+1) * time.Second
 }
```

Comment:
I think using time instead of casting to int / int64 makes this more readable, e.g.:
```
type PostRequestCounter struct {
	Mu           sync.RWMutex
	ReqRemaining int
	RefreshTime  time.Time
}
```

```
func (c *PostRequestCounter) IsPOSTLimitReached() bool {
	// TODO: Once linode API adjusts rate-limits, remove secondary rate limit and simplify accordingly
	return time.Now().Before(c.RefreshTime) && (c.ReqRemaining == 0 || c.ReqRemaining == reconciler.SecondaryPOSTRequestLimit)
}
```

```
func (c *PostRequestCounter) RetryAfter() time.Duration {
	currTime := time.Now()
	if currTime.After(c.RefreshTime) {
		return 0
	}
	return c.RefreshTime.Sub(currTime)
}
```

Reply:
updated code with the suggestion